### PR TITLE
fix: `rpc-proxy` Dockerfile using dependencies from the monorepo

### LINKED ISSUE #1366: 🐛 [BUG] - `rpc-proxy` Docker image does not build when there are breaking changes in the SDK dependencies
### Description

Right now when we build the Docker image via Docker Compose we install and build just the `rpc-proxy` package. 

This means that it is pulling the SDK dependencies from NPM rather than the workspace (local changes). So when there are breaking changes in one of them (like `network`) the image does not build.

cc: @rodolfopietro97 

### Reproduction steps

```bash
1. Go to 'packages/rpc-proxy'
2. Run 'docker-compose up --build rpc-proxy'
```

### PR DESCRIPTION
**Pull request by freemanzMrojo**
# Description

- Some refactor so:
    - All dockerfiles and their config are under the same folder `docker`.
    - `docker-compose` files renamed so they are at the same level as the project root (so same as the parent `package.json`), avoiding with this parent folder issues.

- `rpc-proxy` Dockerfile modified so it pulls SDK dependencies from the monorepo rather than npm.

Closes #1366 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Tested with the SDK test suite and by running from the project root `docker-compose -f docker-compose.rpc-proxy.yml up --build rpc-proxy`.

**Test Configuration**:
* Node.js Version: 20.17.0
* Yarn Version: 1.22.19

# Checklist:

- [x] My code follows the coding standards of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing integration tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

File: docker-compose.rpc-proxy.yml [hunk 1]
```diff
@@ -73,20 +77,24 @@ services:
 
   # 1. Example of a custom configuration file for the RPC proxy service
   rpc-proxy-custom-config-file:
-    build: .
+    build:
```
Comment (leszek-vechain) [inline docker-compose.rpc-proxy.yml:80]: I think examples needs to be commented as if you do docker-compose up it starts everything

Comment (freemanzMrojo) [reply docker-compose.rpc-proxy.yml:80]: It is true what you say, but it was this way before my changes. 

I guess you could run instead `docker-compose -f docker-compose.rpc-proxy.yml up rpc-proxy` or `docker-compose -f docker-compose.rpc-proxy.yml up --build rpc-proxy` if you want to build it. I think this is what @rodolfopietro97 was doing.

Comment (leszek-vechain) [reply docker-compose.rpc-proxy.yml:80]: make sense, but again as lazy user I did just docker-compose up 😄

**Review by rodolfopietro97 — APPROVED** [review]

File: docker-compose.rpc-proxy.yml (hunk 1 above)
Comment (freemanzMrojo) [reply docker-compose.rpc-proxy.yml:80]: Sure, thanks guys for your input. @leszek-vechain if you want as part of a separate ticket we could accommodate the files so they work nicely with `testcontainers` too. We could use docker-compose files to override services so we do not need to comment out anything.

**Pull request merged**